parser/src/main/scala/org/globalnames/parser/ScientificNameParser.scala

```diff
@@ -133,9 +135,10 @@ object ScientificNameParser {
     }
 
     val id: String = {
-      val gn = UUID.fromString("90181196-fecf-5082-a4c1-411d4f314cda")
-      val uuid = UUID.nameUUIDFromString(verbatim, gn, "SHA1").toString
-      s"${uuid.substring(0, 14)}5${uuid.substring(15, uuid.length)}"
+      val namespace = UUID.fromString("90181196-fecf-5082-a4c1-411d4f314cda")
+      Generators.nameBasedGenerator(namespace)
```

Comment:
How is multiple instance creation `Generators.nameBaseGenerator(namespace)` performance wise? Does it make sense to create it once?